src/test/scala/org/hammerlab/guacamole/commands/StructuralVariantCallerSuite.scala

```diff
@@ -1,16 +1,75 @@
 package org.hammerlab.guacamole.commands
 
-import org.hammerlab.guacamole.util.GuacFunSuite
+import org.hammerlab.guacamole.util.{ TestUtil, GuacFunSuite }
 import org.scalatest.Matchers
 
+import scalax.collection.Graph
+import scalax.collection.GraphPredef._
+import scalax.collection.edge.Implicits._
+
 class StructuralVariantCallerSuite extends GuacFunSuite with Matchers {
-  test("coalesce intervals") {
-    val nums = Array(0L, 10L, 20L, 30L,
-      50L,
-      70L, 80L,
-      100L, 110L, 120L,
-      150L)
-    val ranges = StructuralVariant.Caller.coalesceAdjacent(nums, 10)
-    assert(ranges.toList === List((0L, 30L), (50L, 50L), (70L, 80L), (100L, 120L), (150L, 150L)))
+
+  // aliases
+  val MedianStats = StructuralVariant.MedianStats
+  val medianStats = StructuralVariant.Caller.medianStats[Int] _
+
+  test("median stats") {
+    // This example comes from Wikipedia: https://en.wikipedia.org/wiki/Median_absolute_deviation
+    val nums = Array(2, 4, 1, 1, 2, 6, 9)
+    assert(medianStats(nums) == MedianStats(2, 1))
+
+    // An array with an even number of elements an a non-integer median/MAD
+    val nums2 = Array(0, 1, 2, 2)
+    assert(medianStats(nums2) == MedianStats(1.5, 0.5))
+
+    // Pathological cases
+    assert(medianStats(Array(1)) == MedianStats(1.0, 0.0))
+    assert(medianStats(Array()) == MedianStats(0.0, 0.0))
+  }
+
+  sparkTest("read filtering") {
+    val reads = Seq(
+      // A few reads with an insert size of 100 to set the median and MAD
```

Comment:
1. a little confusing that this says 100 but they're all 88 below.
2. does insert size mean "length of first read + the size between the reads"?! that's a 3rd definition that i did not expect.